effectful/handlers/numbers.py

```diff
@@ -102,7 +102,7 @@ def _wrapped_op(x: T_Number) -> T_Number:
 @numbers.Complex.register
 class _ComplexTerm(_NumberTerm, Term[numbers.Complex]):
     def __bool__(self) -> bool:
-        raise ValueError("Cannot convert term to bool")
+        return bool_(self)
```

Comment:
Overloading `__bool__` on terms may not be precise enough to implement the behavior we want.

Reply:
It definitely isn't as precise as we would like. Bool conversion inside terms won't behave correctly.